<a href="https://colab.research.google.com/github/DaianaSep/segundaPreEntrega/blob/main/Segunda_pre_entrega.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Segunda Pre-Entrega: Ingesta, Radiografía y Funciones de Transformación
####**Dataset:** `developer_burnout_dataset_7000.csv`



## BLOQUE 1: INGESTA DE DATOS Y PRIMER VISTAZO

In [9]:
import pandas as pd

# Carga del dataset en un DataFrame llamado df
url = "https://raw.githubusercontent.com/DaianaSep/segundaPreEntrega/refs/heads/main/developer_burnout_dataset_7000.csv"
df = pd.read_csv(url)

# Muestra en pantalla de las primeras 5 filas
display(df.head())


,age,experience_years,daily_work_hours,sleep_hours,caffeine_intake,bugs_per_day,commits_per_day,meetings_per_day,screen_time,exercise_hours,stress_level,burnout_level
0,26.0,12.0,10.33,4.45,2.0,11.0,4.0,1.0,15.07,0.14,55.96,Medium
1,39.0,10.0,8.62,5.77,5.0,15.0,11.0,5.0,13.25,0.54,82.22,High
2,34.0,13.0,NaN,4.03,5.0,2.0,18.0,9.0,11.18,1.54,61.77,Medium
3,30.0,1.0,6.85,6.47,2.0,15.0,26.0,1.0,11.14,0.96,54.98,Medium
4,27.0,7.0,4.24,5.80,NaN,9.0,17.0,7.0,8.05,0.36,27.90,Low


## BLOQUE 2: RADIOGRAFÍA TÉCNICA DEL DATASET


In [10]:
# Dimensiones exactas del dataset (filas y columnas)
print(f"Dimensiones del dataset: {df.shape}")

Dimensiones del dataset: (7000, 12)


In [11]:
# Tipos de datos que procesa cada columna y conteo de no nulos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7000 entries, 0 to 6999
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   age               6860 non-null   float64
 1   experience_years  6860 non-null   float64
 2   daily_work_hours  6860 non-null   float64
 3   sleep_hours       6860 non-null   float64
 4   caffeine_intake   6860 non-null   float64
 5   bugs_per_day      6860 non-null   float64
 6   commits_per_day   6860 non-null   float64
 7   meetings_per_day  6860 non-null   float64
 8   screen_time       6860 non-null   float64
 9   exercise_hours    6860 non-null   float64
 10  stress_level      6860 non-null   float64
 11  burnout_level     6860 non-null   object 
dtypes: float64(11), object(1)
memory usage: 656.4+ KB


In [12]:
# Foto estadística de variables numéricas
display(df.describe())

,age,experience_years,daily_work_hours,sleep_hours,caffeine_intake,bugs_per_day,commits_per_day,meetings_per_day,screen_time,exercise_hours,stress_level
count,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000,6860.000000
mean,32.129446,9.581050,9.000277,6.485638,3.540671,9.482653,14.458455,4.545044,11.974773,1.007611,53.652825
std,7.230254,5.760972,2.863944,1.445008,2.283445,5.751664,8.669846,2.856766,3.091204,0.576708,23.446656
min,20.000000,0.000000,4.000000,4.000000,0.000000,0.000000,0.000000,0.000000,5.210000,0.000000,0.000000
25%,26.000000,5.000000,6.560000,5.220000,2.000000,5.000000,7.000000,2.000000,9.460000,0.510000,36.587500
50%,32.000000,10.000000,8.990000,6.460000,4.000000,9.000000,14.000000,5.000000,12.020000,1.020000,53.795000
75%,38.000000,15.000000,11.422500,7.740000,6.000000,14.000000,22.000000,7.000000,14.422500,1.510000,70.800000
max,44.000000,19.000000,14.000000,9.000000,7.000000,19.000000,29.000000,9.000000,18.930000,2.000000,100.000000


### Hallazgos

#### **1. Anomalía en Edad vs. Experiencia laboral:** A partir del resumen estadístico, identifico una edad mínima de 20 años y una experiencia laboral máxima de 19 años. Aunque es necesario un análisis posterior para confirmar combinaciones específicas, la proximidad de estos límites me sugiere inconsistencias en el registro de los datos.

#### **2. Distribución uniforme de nulos:** El conteo muestra 6.860 observaciones válidas sobre 7.000 en todas las variables. Este 2% constante de datos faltantes podría indicarme un patrón de pérdida de datos homogéneo.

#### **3. Elevada dispersión del estrés percibido:** la variable de nivel de estrés presenta una media de 53.65 puntos con una elevada desviación estándar (23.45), lo que me indica una considerable variabilidad en la percepción del estrés de los encuestados.



## BLOQUE 3: DIAGNÓSTICO DE DATOS FALTANTES


In [ ]:
# Cálculo del porcentaje exacto de datos nulos ordenado de mayor a menor
porcentaje_nulos = (df.isnull().mean() * 100)

# Presentación formateada de la tabla de ranking de nulos
ranking_nulos = pd.DataFrame({'Porcentaje_Nulos_%': porcentaje_nulos})
display(ranking_nulos)


,Porcentaje_Nulos_%
age,2.0
experience_years,2.0
daily_work_hours,2.0
sleep_hours,2.0
caffeine_intake,2.0
bugs_per_day,2.0
commits_per_day,2.0
meetings_per_day,2.0
screen_time,2.0
exercise_hours,2.0


## BLOQUE 4: LÓGICA ALGORÍTMICA Y FUNCIONES PERSONALIZADAS


In [ ]:
# 4.1. FUNCIÓN FORMAL (def)
# Definición de función con estructura condicional (if/elif/else) basada en la regla de negocio para categorizar el nivel de jornada laboral según las horas de trabajo diario.

def categorizar_jornada(horas):
    # Evalúa el rango de horas de trabajo para clasificar la sobrecarga laboral
    if pd.isna(horas):
        return 'Sin Datos'
    elif horas > 10:
        return 'Jornada Excesiva'
    elif horas >= 8:
        return 'Jornada Completa'
    else:
        return 'Jornada Reducida'

# Aplicación de la función formal usando .apply() para generar una nueva columna
df['categoria_jornada'] = df['daily_work_hours'].apply(categorizar_jornada)

# Verificación del resultado junto con datos faltantes
display(df['categoria_jornada'].value_counts(dropna=False))


,count
categoria_jornada,
Jornada Reducida,2750
Jornada Excesiva,2710
Jornada Completa,1400
Sin Datos,140


In [ ]:
# 4.2. FUNCIÓN EXPRESS (lambda)
# Uso función lambda para normalizar la columna 'burnout_level' a minúsculas.

df['burnout_level_minusc'] = df['burnout_level'].apply(lambda x: x.lower() if isinstance(x, str) else x)

# Muestra del resultado transformado
display(df[['burnout_level', 'burnout_level_minusc', 'categoria_jornada']].head())


,burnout_level,burnout_level_minusc,categoria_jornada
0,Medium,medium,Jornada Excesiva
1,High,high,Jornada Completa
2,Medium,medium,Sin Datos
3,Medium,medium,Jornada Reducida
4,Low,low,Jornada Reducida
